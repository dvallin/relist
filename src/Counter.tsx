import * as React from 'react'

export default class Counter extends React.Component<{}, { count: number }> {
    private intervalHandle: number | undefined

    public constructor(props: any) {
        super(props)
        this.state = { count: 0 }
    }

    public render(): JSX.Element {
        return (
            <span>
                {this.state.count} - {this.generateString1()} -{' '}
                {this.generateString2()}
            </span>
        )
    }

    public componentDidMount(): void {
        this.intervalHandle = window.setInterval(
            () =>
                this.setState(prevState => ({
                    count: prevState.count + 1,
                })),
            200
        )
    }

    public componentWillUnmount(): void {
        clearInterval(this.intervalHandle)
    }

    private generateString1(): string {
        return this.state.count.toExponential()
    }

    private generateString2 = () => {
        return this.state.count.toPrecision(18)
    }
}
