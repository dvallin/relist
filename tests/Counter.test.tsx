import * as React from 'react'
import * as ReactDOM from 'react-dom'
import * as TestUtils from 'react-dom/test-utils'
import Counter from '../src/Counter'

describe(Counter.name, () => {
    it('renders title', () => {
        // when
        const element = TestUtils.renderIntoDocument(<Counter />) as Element
        const node = ReactDOM.findDOMNode(element)!

        // then
        expect(node).toMatchSnapshot()
    })
})
