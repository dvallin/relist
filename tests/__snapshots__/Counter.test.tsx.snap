// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders title 1`] = `
<span>
  0
   - 
  0e+0
   -
   
  0.00000000000000000
</span>
`;
